nautobot/extras/signals.py

```diff
@@ -70,29 +79,37 @@ def _handle_changed_object(change_context, sender, instance, raw=False, **kwargs
 
     # Record an ObjectChange if applicable
     if hasattr(instance, "to_objectchange"):
+        # save a copy of this instance's field cache so it can be restored after serialization
+        # to prevent unexpected behavior when chaining multiple signal handlers
+        original_cache = instance._state.fields_cache.copy()
```

Comment:
Would it be preferable to fix the other handlers to not make assumptions about field caching? Not blocking.

Reply:
Calling `instance.refresh_from_db()` does not refresh cached fields on cached related instances 😕